src/core/grpc-server.ts

```diff
@@ -20,7 +21,9 @@ if (process.env.USE_SSL) {
   credentials = grpc.ServerCredentials.createInsecure();
 }
 
-server.addService(CogService, new Cog(ClientWrapper));
+const client = ContactMixin(ClientWrapper);
```

Comment:
Hmm...  I wonder if it's possible to perform the actual mixin "mixing" within the `client-wrapper.ts` itself.  E.g. the final exported class contains all of the methods, but individual groups of methods would come from mixin classes.  

For example...

```typescript
// src/client/client-wrapper.ts pseudo-code

import { ContactAwareMixin } from './mixins/contact-aware.ts';
import { LeadAwareMixin } from './mixins/lead-aware.ts';
class ClientWrapper {
  constructor(auth: grpc.Metadata, clientConstructor = jsforce) {/*...*/}
  /*...*/
}

interface ClientWrapper extends ContactAwareMixin, LeadAwareMixin  {};

applyMixins(ClientWrapper, [ContactAwareMixin, LeadAwareMixin]);

export ClientWrapper as ClientWrapper
```

```typescript
// src/client/mixins/contact-aware.ts pseudo-code

export class ContactAwareMixin {
  public async createContact(contact) {/*...*/}
  /*...*/
}
```

See: https://www.typescriptlang.org/docs/handbook/mixins.html

Reply:
They say that the example from the typescript _official_ are not up to date. Though the `applyMixins` function is looking clean. Even if it's outdated, should we go with it?

Reply:
By go with it. Completely follow their guide.

Reply:
Let's give it a shot and see what happens...? 😬

Reply:
New change pushed. Let me know what you think. If it's good then I will teach @saludangelito on how to convert his work to `mixins`. Thanks for all your help!